fix: NetworkSceneManager add do not load during callback note in xml documentation [MTT-3495]

NoelStephensUnity opened the pull request:
This pertains to #1942 and was determined that the API documentation needs to be updated to let users know not to start new scene events within a scene event callback.  This PR adds the additional XML documentation.

[MTT-3495](https://jira.unity3d.com/browse/MTT-3495)

## Testing and Documentation
- Includes additions to existing public API documentation.

NoelStephensUnity committed:
update

xml documentation to be included in API documentation that one should not try to start a new scene event within a scene event notification callback.

NoelStephensUnity committed:
update

one more location just to make sure.

jeffreyrainy commented on com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs:
```diff
@@ -12,6 +12,7 @@ namespace Unity.Netcode
     /// delegate type <see cref="NetworkSceneManager.SceneEventDelegate"/> uses this class to provide
     /// scene event status.<br/>
     /// <em>Note: This is only when <see cref="NetworkConfig.EnableSceneManagement"/> is enabled.</em><br/>
+    /// <em>*** Do not start new scene events within scene event notification callbacks.</em><br/>
```
Should we have (or is there) any mechanism to detect and prevent such scene events ?

NoelStephensUnity replied:
Why yes there is!
Both `NetworkSceneManager.LoadScene` and `NetworkSceneManager.UnloadScene` return a `SceneEventProgressStatus`.
If you invoke that in the middle of a scene event `NetworkSceneManager` will return `SceneEventProgressStatus.SceneEventInProgress` which tells/reminds the user "You are in the middle of a scene event, let it finish completely before trying to start a new one".
The same thing would happen if you tried to load (n) scenes all at the same time and called `NetworkSceneManager.LoadScene` within some method.  The first call (iteration 0) it would return `SceneEventProgressStatus.Started` and then for every other call (up to n) it would return `SceneEventProgressStatus.SceneEventInProgress`.

So, this is PR is just to remind users when they are looking through the API that you can't start a new scene event until any current scene events are still in progress.

0xFA11 commented on com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs:
```diff
@@ -156,6 +157,7 @@ public class NetworkSceneManager : IDisposable
         /// <b>Alternate Single Event Type Notification Registration Options</b><br/>
         /// To receive only a specific event type notification or a limited set of notifications you can alternately subscribe to
         /// each notification type individually via the following events:<br/>
+
```
nit :P
```suggestion
```

NoelStephensUnity committed:
Update com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs

Co-authored-by: Fatih Mar <mfatihmar@gmail.com>

NoelStephensUnity committed:
fix - NetworkObjectParentingTests.SetParentDirect

This should fix the edge case scenario exposed on Mac Desktop.
This PR originally only modified comments, but it continued to fail in the one spot we were not using a WaitForConditionOrTimeOut.

netcode-ci-service committed:
Merge develop into fix/mtt-3495-networkscenemnager-xml-documentation-add-do-not-load-during-callback-note
